### ISSUE TITLE
New compilers don't like extern crate

### ISSUE THREAD

**Issue by jbaublitz** [stratis-storage/stratisd#1700]
New compilers prefer `use` over `extern crate`.

**Comment by jbaublitz** [comment]
One note is that cargo does not complain if you attach a `#[macro_use]`. I'm interested in if this can be swapped out for `use [CRATE_NAME]` just as a forward looking move.

**Comment by mulkieran** [comment]
See related: https://github.com/stratis-storage/devicemapper-rs/pull/441#issuecomment-480912999.

**Comment by mulkieran** [comment]
@jbaublitz How new are the new compilers and how do they express their preference. Did you find this in some docs, from compiler errors or warnings, or from clippy warnings?

**Comment by jbaublitz** [comment]
This may have been a misunderstanding on my part. The compiler started throwing errors when I added an `extern crate` for testing libcrypsetup-rs. This probably only became a problem because of our linting which I was not running when testing libcryptsetup-rs as a standalone library. I also was confused about it because of the `#[macro_use]` cases in stratisd. I resolved the issues in libcryptsetup-rs and removing the `extern crate libcryptsetup-rs` which resolved the problem. This can be closed I think.
